src/lenskit/splitting/temporal.py

```diff
@@ -94,6 +99,10 @@ def split_global_time(
             tlog.debug("filtering test data for upper bound")
             test = matrix[mask & (ts_col < t2)]
 
+        if filter_test_users:
+            train_users = set(train_build.build().interactions().pandas(ids=True)["user_id"])
+            test = test[test["user_id"].isin(train_users)]
+
         tlog.debug("building training data set")
         train_ds = train_build.build()
```

Comment:
Should move the `train_ds = ` line before the `if` and use that variable in the filtering, so we only build the data set once.